matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/store/db/RealmCryptoStore.kt

```diff
@@ -140,15 +141,13 @@ internal class RealmCryptoStore @Inject constructor(
         newSessionListeners.remove(listener)
     }
 
-    private val monarchyWriteAsyncExecutor = Executors.newSingleThreadExecutor()
-
-    private val monarchy = Monarchy.Builder()
-            .setRealmConfiguration(realmConfiguration)
-            .setWriteAsyncExecutor(monarchyWriteAsyncExecutor)
-            .build()
+    private val lazyRealmConfiguration by lazy {
+        Log.e("!!!", "ensureCryptoMetadataEntity ${Thread.currentThread().name}")
+        ensureCryptoMetadataEntity(realmConfiguration)
+        realmConfiguration
```

Comment:
How can we ensure that in the future `realmConfiguration` is not used directly by mistake? Maybe first add a comment?

Reply:
hmmmmm we could create a `LazyRealmConfiguration` abstraction, what do you think?

also happy to add a comment (and remove the Log)

Reply:
Maybe a comment on `realmConfiguration` to warn developer is enough... This code will be removed in a near future (Crypto Rust SDK).
And yes, please remove the log 👍

Reply:
done https://github.com/vector-im/element-android/pull/4370/commits/d337b4fb5927efbb5889bdee47d777f6e0384d60